pilot/pkg/xds/eds.go

```diff
@@ -172,6 +172,14 @@ func (s *DiscoveryServer) EDSUpdate(clusterID, serviceName string, namespace str
 	inboundEDSUpdates.Increment()
 	// Update the eds data structures and trigger a push.
 	fp := s.edsUpdate(clusterID, serviceName, namespace, istioEndpoints)
+	if !fp {
```

Comment:
Should we handle this case in `workloadEntryHandler` in servicediscovery or for DNS do a full push some thing like this https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/serviceentry/servicediscovery.go#L201?

Doing it in Eds, would involve `GetService` call for every K8s endpoint also. 

If doing it in workloadEntryHandler is complicated - may be we should check clusterID and do only workload entries?

Reply:
Agree, it seems it should be fixed in `serviceEntryHandler`. I will try tomorrow.